statvar_imports/us_bls/cpi_category/cpi_category_download.py

```diff
@@ -0,0 +1,419 @@
+                            else:
+                                logging.debug(f"Skipping {file_name} from zip (cpi-u/w, not January or specific latest {latest_year}).")
+
+                        elif category_folder_name == "c-cpi-u":
+                            # Apply c-cpi-u rules: even months from Dec 2021 onwards
+                            start_year_ccpiu = c_cpi_u_rules['start_year']
+                            if year >= start_year_ccpiu:
+                                if year == start_year_ccpiu and month == 12: # Dec 2021
+                                    move_file = True
+                                elif year > start_year_ccpiu and month % 2 == 0: # Even months for subsequent years
+                                    move_file = True
+                                else:
+                                    logging.debug(f"Skipping {file_name} from zip (c-cpi-u, not matching even month/start year rule).")
+                            else:
+                                logging.debug(f"Skipping {file_name} from zip (c-cpi-u, year older than {start_year_ccpiu}).")
+                        else:
+                            logging.debug(f"Skipping {file_name} from zip (unrecognized CPI category).")
+                    else:
+                        logging.debug(f"Skipping {file_name} from zip (could not extract year/month).")
+
+
+                    if move_file:
+                        source_path = os.path.join(root, file_name)
+                        destination_category_path = os.path.join(final_destination_base_path, category_folder_name)
+                        os.makedirs(destination_category_path, exist_ok=True)
+                        destination_file_path = os.path.join(destination_category_path, file_name)
+
+                        try:
+                            # Overwrite if file already exists (e.g., from individual download or duplicate in zip)
+                            if os.path.exists(destination_file_path):
+                                logging.warning(f"File '{file_name}' already exists in '{destination_category_path}'. Overwriting.")
+
+                            shutil.move(source_path, destination_file_path)
+                            logging.info(f"Moved CPI file '{file_name}' from '{source_path}' to '{destination_category_path}'")
+                            found_cpi_files = True
+                        except OSError as e:
+                            logging.error(f"Error moving file {file_name} from {source_path} to {destination_file_path}: {e}")
+                # else:
+                #     logging.debug(f"Skipping non-Excel file: {file_name} from zip.")
+
+        if not found_cpi_files:
+            logging.info(f"No *relevant* CPI Excel files found in {os.path.basename(zip_file_path)} based on specified rules.")
+
+    except zipfile.BadZipFile:
+        logging.error(f"Error: {os.path.basename(zip_file_path)} is not a valid ZIP file.")
```

Comment:
these errors will not throw exception upstream. is this intended? should the process stop if unzipping fails? if yes, is it handeled?

Reply:
Changes are done. If unzipping fails, the script will trigger logging.fatal.

Reply:
```suggestion
        logging.error(f"Error: {os.path.basename(zip_file_path)} is not a valid ZIP file.")
```

log execption as well.

Reply:
Added TODO in code and README